### PR TITLE
Clarify how auto should be resolved

### PR DESCRIPTION
**Pull request by simonrankine**

### CODE REVIEW

File: docs/2.2. APIs - Server Side Implementation.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ When a Sender or Receiver is first started it may not have all the parameters it
 However Senders and Receivers may start with parameters already configured, for example through use of a configuration file or manual entry of parameters.
 
 ## Use of auto
-Many transport parameters in the API may be set to "auto" in the `/staged` endpoint, to imply the Sender or Receiver should select that parameter for itself. In many cases this is a simple operation, and the behaviour is very clearly defined in the relevant transport parameter schemas. For example a port number may be offset from the RTP port number by a pre-determined value.
+Many transport parameters in the API may be set to "auto" in the `/staged` endpoint, to imply the Sender or Receiver should select that parameter for itself. In many cases this is a simple operation, and the behaviour is very clearly defined in the relevant transport parameter schemas. For example a port number may be offset from the RTP port number by a pre-determined value. The specification makes suggestion makes suggestions of a sensible value for auto to resolve to, but the Sender or Receiver may choose any value permitted by the schema.
```
Comment (andrewbonney) [inline docs/2.2. APIs - Server Side Implementation.md]:
```suggestion
Many transport parameters in the API may be set to "auto" in the `/staged` endpoint, to imply the Sender or Receiver should select that parameter for itself. In many cases this is a simple operation, and the behaviour is very clearly defined in the relevant transport parameter schemas. For example a port number may be offset from the RTP port number by a pre-determined value. The specification makes suggestions of a sensible value for auto to resolve to, but the Sender or Receiver may choose any value permitted by the schema.
```

**Review by garethsb — CHANGES_REQUESTED** [review]
Thanks, Simon. I think the intent is clear and I agree with the strategy (though it means implementation work for me, sigh).

Unfortunately, I think we also need to modify the ``"description"`` strings in the schema files as well because things like "(auto = 5004)" don't indicate the implementation flexibility. Maybe adding " by default" to all of them would be enough...

**Comment by andrewbonney** [comment]
It may be worth an additional clarification in here noting that 'auto' doesn't need to be included in the constraints as it's defined in the schemas for the transport types.

**Review by garethsb — APPROVED** [review]
LGTM.

**Review by garethsb — COMMENTED** [review]
LGTM.

**Pull request merged**